### PR TITLE
Add logic to diff two coverage reports

### PR DESCRIPTION
**Pull request by mckornfield**

### CODE REVIEW

**Comment by mckornfield** [comment]
@noahdenier Ready for review

**Comment by mckornfield** [comment]
Pushed up some extra coverage and a bugfix as well. If we want we can add test coverage automatically via travis ci too.

File: test/converterTest.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+})
+
+function assertDiffNamePresentAndLineNumbers(diff, name, lineNumbers) {
+  diffElement = diff.find((elem) => {
+    return elem.name === name
+  })
+  should.exist(diffElement, `${name} not found in diff`)
+  expect(diffElement.line).to.eql(lineNumbers)
+}
+
```
Comment (noahdenier) [inline test/converterTest.js]: Test code looks great! 😄

File: src/converter.js [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+function diffCoverage(jsonObj1, jsonObj2) {
+  mapping1 = createSourceFileNameToDataMap(jsonObj1)
+  mapping2 = createSourceFileNameToDataMap(jsonObj2)
+  diff = []
+  for (key in mapping1) {
+    if (!(key in mapping2)) {
+      diff.push({
+        name: key,
+        line: mapping1[key]['line'].map(line => parseInt(line.nr)),
```
[the remark this answers is not in the document]
Comment (mckornfield) [reply src/converter.js]: I think I used it right but might be wrong, it's just supposed to get all the line numbers from that array

File: src/converter.js [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+  line2Map = createLineNumberToLineMap(data2)
+  lineDiff = []
+  for (lineNumber in line1Map) {
+    if (!(lineNumber in line2Map)) {
+      lineDiff.push(parseInt(lineNumber))
+    } else {
+      coveredInstruction1 = line1Map[lineNumber]
+      coveredInstruction2 = line2Map[lineNumber]
+      // Both lines present, check to see if instructions are not covered by automated test
+      // TODO: Check if this is sufficient
```
[the remark this answers is not in the document]
Comment (mckornfield) [reply src/converter.js]: will try a different structure. I think the original will just have the covered lines in it

File: src/converter.js [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+  }
+  return lineDiff
+}
+
+function createLineNumberToLineMap(lineData) {
+  let mapping = {}
+  console.log(lineData)
+  lineDataAsArray = Array.isArray(lineData) ? lineData : [lineData]
+  lineDataAsArray.forEach((obj) => {
+    mapping[obj.nr] = obj.ci
```
[the remark this answers is not in the document]
Comment (mckornfield) [reply src/converter.js]: If it is, haven't seen it yet

File: src/converter.js [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+  })
+  return mapping
+}
```
[the remark this answers is not in the document]
Comment (mckornfield) [reply src/converter.js]: F that JS linter I do what I want

File: test/converterTest.js (hunk 1 above)
Comment (mckornfield) [reply test/converterTest.js]: yayyy

**Comment by mckornfield** [comment]
@noahdenier ready for another round 🥊

File: src/diff.js [hunk 6]
```diff
@@ -0,0 +1,40 @@
+module.exports = { diffCoverage }
+
+
+function diffCoverage(jsonObj1, jsonObj2) {
+    mapping1 = createSourceFileNameToDataMap(jsonObj1)
+    mapping2 = createSourceFileNameToDataMap(jsonObj2)
```
Comment (noahdenier) [inline src/diff.js]: This is good for now, going to make a note that maybe we should move this transformation to the converter, and the diff can just assume it's in this structure

**Pull request merged**